src/libraries/Common/src/System/Security/Cryptography/SlhDsa.cs

```diff
@@ -0,0 +1,1276 @@
+            {
+                throw new ArgumentException(nameof(destination), SR.Argument_DestinationTooShort);
+            }
+
+            ExportSlhDsaPublicKeyCore(destination.Slice(0, publicKeySizeInBytes));
+            return publicKeySizeInBytes;
+        }
+
+        /// <summary>
+        ///  Exports the current key in the FIPS 205 secret key format.
+        /// </summary>
+        /// <param name="destination">
+        ///   The buffer to receive the secret key.
+        /// </param>
+        /// <returns>
+        ///   The number of bytes written to <paramref name="destination"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="destination"/> is too small to hold the secret key.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while exporting the key.
+        /// </exception>
+        public int ExportSlhDsaSecretKey(Span<byte> destination)
+        {
+            ThrowIfDisposed();
+
+            int secretKeySizeInBytes = Algorithm.SecretKeySizeInBytes;
+
+            if (destination.Length < secretKeySizeInBytes)
+            {
+                throw new ArgumentException(nameof(destination), SR.Argument_DestinationTooShort);
+            }
+
+            ExportSlhDsaSecretKeyCore(destination.Slice(0, secretKeySizeInBytes));
+            return secretKeySizeInBytes;
+        }
+
+        /// <summary>
+        ///  Exports the private seed of the current key.
+        /// </summary>
+        /// <param name="destination">
+        ///   The buffer to receive the private seed.
+        /// </param>
+        /// <returns>
+        ///   The number of bytes written to <paramref name="destination"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="destination"/> is too small to hold the private seed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while exporting the private seed.
+        /// </exception>
+        public int ExportSlhDsaPrivateSeed(Span<byte> destination)
+        {
+            ThrowIfDisposed();
+
+            int privateSeedSizeInBytes = Algorithm.PrivateSeedSizeInBytes;
+
+            if (destination.Length < privateSeedSizeInBytes)
+            {
+                throw new ArgumentException(nameof(destination), SR.Argument_DestinationTooShort);
+            }
+
+            ExportSlhDsaPrivateSeedCore(destination.Slice(0, privateSeedSizeInBytes));
+            return privateSeedSizeInBytes;
+        }
+
+        /// <summary>
+        ///   Generates a new SLH-DSA key for the specified algorithm.
+        /// </summary>
+        /// <returns>
+        ///   The generated object.
+        /// </returns>
+        public static SlhDsa GenerateKey(SlhDsaAlgorithm algorithm) =>
+            new SlhDsaImplementation(algorithm);
```

Comment:
A) the argument validation should happen here, at the public method.
B) it shouldn't be using a ctor for SlhDsaImplementation, but calling (e.g.) `internal static SlhDsaImplementation.GenerateKeyCore(algorithm)` (since "GenerateKey" is taken by this method).  The ctors should only accept an already generated key handle.

Reply:
The base class doesn't have handles, so it needs to be told what the algorithm is so it can put it on the property and use it for length checks.

SlhDsaImplementation will probably have a ctor that takes both the handle for the current platform (to store as a field) and an algorithm ID (to pass down to the base ctor).  It won't have one that takes just an algorithm.

Reply:
e.g.

```C#
internal static SlhDsaImplementation GenerateKeyCore(SlhDsaAlgorithm algorithm)
{
    return new SlhDsaImplementation(
        algorithm,
        Interop.Crypto.GenerateSlhDsaKey(algorithm.Name));
}
```

Reply:
Updated with the actual `GenerateKeyCore` implementation stubbed out.